### PR TITLE
WFLY-19500 Replace deprecated calls to the JBoss Logger methods

### PR DESCRIPTION
**Pull request by marko-bekhta**
https://issues.redhat.com/browse/WFLY-19500

Now, this one is only a draft since it requires a PR for the core to be merged and released (https://github.com/wildfly/wildfly-core/pull/6058).

Same as in the core PR `wildfly-maven-plugin` requires an override of jboss-logging to make things pass the simple build.

____

<--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->

> Wildfly issue links:
> * [WFLY-19684](https://issues.redhat.com/browse/WFLY-19684)

<--- END OF WILDFLY GITHUB BOT REPORT --->

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

### CODE REVIEW

File: build/pom.xml [hunk 1]
```diff
@@ -103,6 +103,14 @@
...
                     </execution>
                 </executions>
+                <!-- Need to override the plugin dependency to make jboss-logging lib work -->
+                <dependencies>
+                    <dependency>
+                        <groupId>org.jboss.logging</groupId>
+                        <artifactId>jboss-logging</artifactId>
+                        <version>${version.org.jboss.logging.jboss-logging}</version>
+                    </dependency>
+                </dependencies>
```
Comment (jamezp) [inline build/pom.xml:113]: We should be able to remove this now.

File: ee-build/pom.xml [hunk 2]
```diff
@@ -94,6 +94,14 @@
...
                     </execution>
                 </executions>
+                <!-- Need to override the plugin dependency to make jboss-logging lib work -->
+                <dependencies>
+                    <dependency>
+                        <groupId>org.jboss.logging</groupId>
+                        <artifactId>jboss-logging</artifactId>
+                        <version>${version.org.jboss.logging.jboss-logging}</version>
+                    </dependency>
+                </dependencies>
```
Comment (jamezp) [inline ee-build/pom.xml:104]: Same comment about removing this :)

File: pom.xml [hunk 3]
```diff
@@ -321,12 +321,17 @@
         <version.org.jboss.wildscribe>3.1.0.Final</version.org.jboss.wildscribe>
         <version.org.wildfly.bom-builder-plugin>2.0.6.Final</version.org.wildfly.bom-builder-plugin>
         <version.org.wildfly.checkstyle-config>1.0.8.Final</version.org.wildfly.checkstyle-config>
-        <version.org.wildfly.common>1.6.0.Final</version.org.wildfly.common>
+        <version.org.wildfly.common>2.0.0</version.org.wildfly.common>
```
Comment (jamezp) [inline pom.xml:324]: We should remove this upgrade. It will be a lot more involved.

File: pom.xml [hunk 4]
```diff
@@ -321,12 +321,17 @@
...
         <version.org.wildfly.unstable.annotation.api>1.0.0.Beta1</version.org.wildfly.unstable.annotation.api>
         <version.org.wildfly.galleon-plugins>7.1.0.Final</version.org.wildfly.galleon-plugins>
         <version.org.wildfly.jar.plugin>11.0.2.Final</version.org.wildfly.jar.plugin>
         <version.org.wildfly.licenses.plugin>2.4.1.Final</version.org.wildfly.licenses.plugin>
         <version.org.wildfly.plugin>5.0.0.Final</version.org.wildfly.plugin>
+        <!--
+            This jboss-logging version is only used to override the version in the wildfly plugin!
+            Remove it as soon as the plugin is updated to a compatible version.
+         -->
+        <version.org.jboss.logging.jboss-logging>3.6.0.Final</version.org.jboss.logging.jboss-logging>
```
Comment (jamezp) [inline pom.xml:334]: We can remove this too.

File: pom.xml [hunk 5]
```diff
@@ -1395,6 +1400,19 @@
...
+                    <version>${version.org.wildfly.plugin}</version>
+                    <!-- Need to override the plugin dependency to make jboss-logging lib work -->
+                    <dependencies>
+                        <dependency>
+                            <groupId>org.jboss.logging</groupId>
+                            <artifactId>jboss-logging</artifactId>
+                            <version>${version.org.jboss.logging.jboss-logging}</version>
+                        </dependency>
+                    </dependencies>
+                </plugin>
```
Comment (jamezp) [inline pom.xml:1415]: We should be able to remove this as it was fixed in Galleon.

File: preview/build/pom.xml [hunk 6]
```diff
@@ -93,6 +93,14 @@
...
                     </execution>
                 </executions>
+                <!-- Need to override the plugin dependency to make jboss-logging lib work -->
+                <dependencies>
+                    <dependency>
+                        <groupId>org.jboss.logging</groupId>
+                        <artifactId>jboss-logging</artifactId>
+                        <version>${version.org.jboss.logging.jboss-logging}</version>
+                    </dependency>
+                </dependencies>
```
Comment (jamezp) [inline preview/build/pom.xml:103]: Same comment about removing this :)

File: testsuite/integration/smoke/pom.xml [hunk 7]
```diff
@@ -538,6 +538,14 @@
...
                     </execution>
                 </executions>
+                <!-- Need to override the plugin dependency to make jboss-logging lib work -->
+                <dependencies>
+                    <dependency>
+                        <groupId>org.jboss.logging</groupId>
+                        <artifactId>jboss-logging</artifactId>
+                        <version>${version.org.jboss.logging.jboss-logging}</version>
+                    </dependency>
+                </dependencies>
```
Comment (jamezp) [inline testsuite/integration/smoke/pom.xml:548]: Same comment about removing this :)

**Review by jamezp — CHANGES_REQUESTED** [review]
We're waiting for the core upgrade, but here is a preliminary review.

File: testsuite/integration/smoke/pom.xml (hunk 7 above)
Comment (marko-bekhta) [reply testsuite/integration/smoke/pom.xml:548]: Thanks for taking a look 😃 
I'll wait for the core release and then apply the changes 🙈

Comment (marko-bekhta) [reply testsuite/integration/smoke/pom.xml]: ohh I didn't realize that it all went so fast and that the core is already out 😃 
I've rebased this one on top of  https://github.com/wildfly/wildfly/pull/18156

**Review by jamezp — APPROVED** [review]

**Review by jamezp — APPROVED** [review]

**Comment by marko-bekhta** [comment]
rebased, as core PR was merged.

**Review by jasondlee — APPROVED** [review]

**Review by jamezp — APPROVED** [review]

**Comment by marko-bekhta** [comment]
Thanks James ! 
ORM, Search and Validator upgrades are next in line 😃 🤞🏻

**Comment by jamezp** [comment]
Thank you @marko-bekhta!

**Pull request merged**